### DISCUSSION TITLE
2021-12-20 planning (covers two weeks)

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#905] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

**This discussion covers planning for two weeks. Most of the team will be off until the new year starting Friday, so there will be no post next week.**

## Discussion topics
1. [Frontend priorities](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1844349)
2. [Backend priorities](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1844362)
3. [Design priorities](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1844375)
4. [Product priorities](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1844376)
5. [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1844382)
6. [Kriti's check in](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1844393)
7. [Dominykas' check-in](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1846136)
8. [Sean's check-in](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1847288)
9. [Brent's check-in](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1847370)
10. [Pavish's check-in](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1847473)
11. [Ghislaine's check-in](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1849163)
12. [Mukesh's check-in](https://github.com/centerofci/mathesar/discussions/905#discussioncomment-1850194)

## Weekly check-in
@centerofci/mathesar-core please respond to the weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

- How did the last week go? What went well, what could have gone better?
- What are you planning to work on over the next two weeks? Do you anticipate any blockers?
- Anything else you'd like to add?

**Comment by kgodey** [comment]
## Frontend priorities

1. Frontend changes to unblock merge of #862
2. Review and merge PRs waiting for review. 
    1. #857 
    2. #861 
    3. #870 
    4. #886 
    5. #904
3. Continue to make progress on the [Working with Tables](https://github.com/centerofci/mathesar/milestone/8) milestone. There only seem to be two unblocked issues.
    1. #500 
    2. #531
4. If Make progress on the [2021-12 Improvements](https://github.com/centerofci/mathesar/milestone/32) milestone if other issues remain blocked. Only two frontend issues remain to be started. We can move these out to a later milestone if needed.
    1. #475 
    2. #632

@pavish @seancolsen Any concerns?

**Reply by seancolsen** [reply]
Looks good :grin:

**Reply by pavish** [reply]
Sounds good to me!

@kgodey I think we can move https://github.com/centerofci/mathesar/issues/632 to a later milestone. It's not a priority and a good first issue, something we can point new contributors to.

**Reply by kgodey** [reply]
@pavish Thanks, I moved #632 to the March milestone.

**Comment by kgodey** [comment]
## Backend priorities

1. Get the following PRs merged into `master`. Please be proactive about communicating blockers for yourself and others.
    1. #837 
    2. #862 
    3. #865
    4. #901
2. Infrastructural issues for API error handling: 
    1. #859 (top priority)
    2. #623
    2. #883 
    2. #889
3. Once the basic grouping and filtering PRs are merged, we should continue to get currently-blocked open PRs for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) merged
    1. #879 
    2. #885 
    3. #893
4. It would be great if someone could take these two open PRs and get them merged. They are both reviewed and waiting for updates from a contributor, but we haven't heard from them for a while. They should be fairly minor.
    1. #797 
    2. #817 
5. Continue to make progress on other [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)  issues.

@dmos62 @mathemancer @silentninja Any concerns?

**Reply by dmos62** [reply]
To get #837 ready for merge, I'll either cherrypick from #862 or merge it in. Either way, #862 should be merged into master first.

**Reply by kgodey** [reply]
Go ahead and merge #862 into your branch, I agree it should be merged to `master` first.

**Reply by mathemancer** [reply]
I think this makes sense.  I have a few more grouping changes to make in a next PR once the current one is merged (I wanted to stop growing the added features at some point).

**Reply by silentninja** [reply]
I can work PR #797 and #817, but I think it would be better if we wait until the holidays are over. The priorities look good otherwise.

**Reply by kgodey** [reply]
@silentninja Why wait until the holidays are over? You should be able to test them on the branch that they are in and queue them up for review and merge once most of us are back.

**Reply by silentninja** [reply]
@kgodey Some contributors might be finding time during the holidays, so want to wait for a reply until then.

**Comment by kgodey** [comment]
## Design priorities

1. We've been through another round of review on the open design specs and should be close to merging them.
    1. https://github.com/centerofci/mathesar-wiki/pull/13
    2. https://github.com/centerofci/mathesar-wiki/pull/12
2. It would be great to get some design discussions started for these issues:
    1.  https://github.com/centerofci/mathesar/issues/455
    2. https://github.com/centerofci/mathesar/issues/466

@ghislaineguerin @ppii775 Any concerns or questions?

**Reply by ppii775** [reply]
@kgodey @ghislaineguerin All good on my end for now! I'm preparing two discussion items (viewing a view – refresh, viewing a view – column provenance and data types) for #466. This semester has ended last Fri and I have free time next week – feel free to reach out! However, I will be offline from Dec 24 to Jan 8. Have a great holiday ahead everybody!

**Reply by ghislaineguerin** [reply]
@kgodey no concerns. looks good

**Comment by kgodey** [comment]
## Product priorities

1. Define Mathesar Types better: https://github.com/centerofci/mathesar-wiki/issues/14
2. Define data sources in Views better and update other Views tickets: #806
3. Think through representing filtering based on CTEs to users (and when that would make sense in our roadmap), see:
    1. #846 
    2. #853 
4. Collaborate with @seancolsen to make issues for new design specs once merged.

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2021-12-20 to 2021-12-23 | Dominykas |
| 2021-12-24 to 2022-01-02 | Pavish |
| 2022-01-03 to 2022-01-07 | Ghislaine |
| 2022-01-10 to 2022-01-14 | Brent |
| 2022-01-17 to 2022-01-21 | Kriti |
| 2022-01-24 to 2022-01-28 | Pavish |
| 2022-01-31 to 2022-02-04 | Mukesh |
| 2022-02-07 to 2022-02-11 | Sean |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by kgodey** [comment]
## Kriti's check in

> How did the last week go? What went well, what could have gone better?

My check in last week was pretty late in the week and not much has changed since then. I am caught up on various reviews and I also caught up on some miscellaneous tasks that didn't require too much focus like fixing the staging server deployment and some workflow automation. I've been putting those off for a while so that was nice. 

> What are you planning to work on over the next two weeks? Do you anticipate any blockers?

I'm off starting Friday so it's only four days for me. I hope to have some focused time to get some product work done. I also want to do some wiki updates, more workflow automation, and fix the inference issues on the staging server if I don't have the right focus for product work.

**Comment by dmos62** [comment]
> How did the last week go? What went well, what could have gone better?

Pretty good. Made progress with filtering. Factored out `sqlalchemy-filters`, which I was worried would be complicated (wasn't).

> What are you planning to work on over the next two weeks? Do you anticipate any blockers?

I want to get the first filtering PR merged. Then continue working through filtering for different types. Due to recent refactors that should be easy. I'm the current comms assignee, which might be interesting.

**Comment by seancolsen** [comment]
## Sean's check in

How did the last week go? What went well, what could have gone better?

- I appreciated the sync call on range grouping. Even though it took a while, I think it was a good use of time.
- I should have been more vocal about the fact that my changes in #836 also fixed a bug preventing users from deleting tables. Had I given others a heads up, I would have saved Pavish and Gabe some work in #887 and #888.
- I think we should all make an effort to move technical conversations out of private channels when they arise there. For example, [here](https://github.com/centerofci/mathesar/issues/895#issuecomment-996720896) I linked to a Matrix thread in a GitHub issue, but the thread is in a private channel, making it hard for external contributors to follow the conversation. Not a huge deal, but it's also pretty easy to switch to the Code channel, so I'll try to be more mindful of that.
    - As an aside, I really wish the GitHub comments functionality had more features to foster realtime conversation like a chat system. It's happened several times that someone has submitted a comment they might have written differently had they seen the latest comment posted by someone else while they were writing. I wish GitHub would load comments automatically and also show me when others are typing. To me, GitHub doesn't work well enough for synchronous conversations because the feedback loop is too slow when others are responding -- and Matrix doesn't work well enough for async conversations because its threading model is weak. I've heard good things about [Zulip](https://zulip.com/) being a platform that strikes the right balance between sync and async communication, but I've never used it. This is just food for thought. I'm not advocating for any particular action right now -- just putting these opinions out there for others to consider.

What are you planning to work on over the next two weeks? Do you anticipate any blockers?

- Add front end changes to the [Range Grouping](https://github.com/centerofci/mathesar/pull/862) PR
- Review [Add documentation for pagination component](https://github.com/centerofci/mathesar/pull/886)
- [Frontend work for adding multi-column unique constraints](https://github.com/centerofci/mathesar/issues/531)
- I should hopefully have some more time after that stuff, in which case I'll work on issues in "2021-12 Improvements".

Anything else you'd like to add?

- My kids are on winter break from school starting today and they're home with my wife Libby watching them which will be a bit of a test of how well she is recovering from her back injury. So I'll probably be in and out of work a bit more than usual this week, and I may put in some hours next week to make up for any shortages I incur this week.

**Reply by kgodey** [reply]
> I think we should all make an effort to move technical conversations out of private channels when they arise there. 

I second this.

> To me, GitHub doesn't work well enough for synchronous conversations because the feedback loop is too slow when others are responding -- and Matrix doesn't work well enough for async conversations because its threading model is weak.

I agree wholeheartedly. The Matrix / Element team is working on threading support, hopefully it will be launched soon and work well enough to make our life easier. See https://github.com/vector-im/element-web/issues/19270.

**Comment by mathemancer** [comment]
## Brent's Check in

> How did the last week go? What went well, what could have gone better?

Things went pretty well, but merging the grouping PR took longer than expected.  We kept going in circles on the API form; perhaps I should have called the meeting to sort that out earlier in the process.

> What are you planning to work on over the next two weeks? Do you anticipate any blockers?

I'll be out next week for the most part, as well as a short week this week.  I'm primarily focused on getting all PRs reviewed and merged in the back end right now, since I suspect coming back to those after the break would be painful.  I'm also going to add another grouping type if there's time, and work on the Boolean type if I want a shorter project.

> Anything else you'd like to add?

Nothing at the moment.

**Comment by pavish** [comment]
## Pavish's check in

> How did the last week go? What went well, what could have gone better?

It went pretty good. I reviewed and merged a number of PRs. I put some more thought into dynamic forms, type configuration layout and cellInput. It took me a while to figure out a good pattern, I did not get enough time to code.

> What are you planning to work on over the next two weeks? Do you anticipate any blockers?

I'm planning to raise PRs for the following:
* Form builder
   - Generate dynamic forms based on a json configuration. This is required before we can implement layout generation for types and cell input. This would also be the core for when we have the form view for table data entry.
* Fix npm audit issues for packages which can be safely fixed. I've been seeing a number of comments by the bot cluttering our github channel on matrix.
* Type configuration layout
   - Generate the type selection layout dynamically using a json configuration. Also, raise a discussion on the format.
   - The initial PR will cover Text, Number and Boolean, and can be considered partial work on the data types milestone.
* CellInput component
   - The input component for table (spreadsheet? - not sure what we should call our layout) cell.
   - The initial component will cover text, textArea, integer, float, checkbox
* Set default value for column: #500 
   - All the above mentioned PRs are per-requisites for this PR. I've not created individual issues for the above (except npm audit failue) because all of them need to be implemented one by one in a synchronous manner as they depend on one another. I'll be raising discussions for those which require them.

Apart from this, I have a bunch of PRs to review and merge.

**Comment by ghislaineguerin** [comment]
## Ghislaine's Check-In

> How did the last week go? What went well, what could have gone better?

It went well. I had focused time to work on the prototype and update some UI components needed to support the listing of views design.

> What are you planning to work on over the next two weeks? Do you anticipate any blockers?

Review design work for 'Viewing Views' and left pane improvements

> Anything else you'd like to add?

Nothing at the moment.

**Comment by silentninja** [comment]
## Mukesh's Check in

> How did the last week go? What went well, what could have gone better?

It went well, glad we were able to finalize the grouping api.  Had to drop the work on data types as there were a few issues blocking others and decided to work on those instead. 

> What are you planning to work on over the next two weeks? Do you anticipate any blockers?

Review and approve the pending filtering and grouping PR's and work on Replacing column names/index usage with column id #839 and related issues as they are blocking the frontend. I will also be working on completing the requested review changes for TImetamp PR #865 and Handling error consistently PR #909 

> Anything else you'd like to add?

Nothing at the moment.

**Reply by kgodey** [reply]
I don't think #839 is blocking the frontend, @seancolsen mentioned in #870 that he's not blocked. I would prioritize #865 and error handling over that.

**Reply by silentninja** [reply]
Thanks for reminding, forgot about that conversation. Will prioritize #865
